pkg/cloud/isolated_network.go

```diff
@@ -44,6 +44,7 @@ type IsoNetworkIface interface {
 func (c *client) getOfferingID() (string, error) {
 	offeringID, count, retErr := c.cs.NetworkOffering.GetNetworkOfferingID(NetOffering)
 	if retErr != nil {
+		c.customMetrics.EvaluateErrorAndIncrementAcsReconciliationErrorCounter(retErr)
```

Comment:
Yeah, I think we can add a function called `reportError` or something and instead of the above we can call:
``` Go
		return "", reportErr(retErr)
```

Reply:
I'm not a fan of reportErr, because it doesn't really tell me what it's going to do. As a frequent code reader, I value explicitness over terseness.  I could have that function return the err that was passed in, so it could be placed in the return statement instead of on its own line.  I'm not feeling, though, like we really made a difference to readability/maintainability.  Same function call, just in a different line.

As the name EvaluateErrorAndIncrementAcsReconciliationErrorCounter implies, I sought get rid of *if err* conditional code.  (See [affinitygroups.go line 96](https://github.com/kubernetes-sigs/cluster-api-provider-cloudstack/blob/5efc0bc760cbf5df0181829b7f21d29bd11da69e/pkg/cloud/affinity_groups.go#L96)).

But our error handling patterns vary a bit.  Sometimes there's need for an else, so the if is still required.  Sometimes there's wrapping in Multierrors after the call.  

There were cases where I was tempted to re-structure the error handling, but ultimately decided the safe road of least-impact was the way to go.

After wrangling the variations, I concluded that this form pretty universally served all the situations without requiring variations, and went with it.